### PR TITLE
Add custom jwt flow

### PR DESCRIPTION
**Pull request by sirewix**
Closes https://github.com/famedly/product-management/issues/1910

### CODE REVIEW

**Review by emgrav — APPROVED** [review]
Have you had a chance to manually test the flow against synapse? If not, please just do that to double check before merging

**Comment by sirewix** [comment]
@emgrav checked, seems to be working. Also added bearer access token (configurable via config) for the server that has to be notified on user registration (forgot about it)

File: synapse_token_authenticator/config.py [hunk 1]
```diff
@@ -46,25 +46,45 @@ def __init__(self, other: dict):
...
-            "ES256",
-            "ES384",
-            "ES512",
-            "PS256",
-            "PS384",
-            "PS512",
-            "EdDSA",
-        ]:
-            raise Exception(f"Unknown algorithm: '{self.jwt.algorithm}'")
+        if custom_flow := other.get("custom_flow"):
```
Comment (emgrav) [inline synapse_token_authenticator/config.py:49]: All the new config options should be documented in the README.

**Review by emgrav — CHANGES_REQUESTED** [review]
Looks great! I think this is pretty much good to go, there's just some documentation I think should be added. 

Additionally, we have the details of the flow documented internally, would it make sense to add a brief and generic description of the new flow in the README as well?

**Review by emgrav — APPROVED** [review]

**Pull request merged**